### ISSUE TITLE
CSV import issue

### ISSUE THREAD

**Issue by hubortje** [helloworld1/AnyMemo#505]
Hello,

I tried importing flashcards from cram.com via .csv-import, but the .db-file is empty after the import.

e.g. https://www.cram.com/flashcards/export/6994298 -> export to CSV
When moving the .csv-file to my mobile and import it via AnyMemo, the .db-file becomes empty. The .csv-file contains entries.

Thank you already!

**Comment by helloworld1** [comment]
Thanks for reporting this issue. I am able to reproduce the issue. First the downloaded CSV file is not actually CSV (Comma separated values), it is separated by tabs. AnyMemo supports tab separated file but only with .txt.

Second, even I renamed the file to .txt, I still get errors. I will investigate the error.

**Comment by hubortje** [comment]
Thank you very much for your quick reply.
Yeah, I thought that might be a problem - I'll try to import a comma separated CSV file, and will report back here.

**Comment by helloworld1** [comment]
Did you try the CSV files exported from AnyMemo? Looks like it can be imported successfully. So at least, some properly formatted CSV file works fine.

**Comment by hubortje** [comment]
Read this just now.
No, I did not try a CSV exported from AnyMemo. As I said, I tried using a flashcards set from cram.com, as they had one for Cantonese.
What is a "properly formatted" CSV for you, because as far as I understand, there is no real standard for formatting CSV-files. What kind of formatting does AnyMemo expect? If you'd have an example, I could at least try to reformat the one exported from Cram to fit the system of AnyMemo.

Thanks for your help!

**Comment by helloworld1** [comment]
I tried it out and the CSV file seems to be valid and importable by AnyMemo. But the import process failed. It is likely a bug in AnyMemo that is not compatible with Android 11 very well. I will investigate further and report back.

**Comment by hubortje** [comment]
Thank you very much!

**Comment by hubortje** [comment]
Quick update:

I'm using Anymemo quite frequently, since October 2021.
I had no issues with csv-files since then, up until today, so I've investigated further.

I can now say for sure that Anymemo expects some kind of formatting for CSV-files, and Microsoft Excel's standard formatting is a huge issue (at least in my experience).
Excel does not place "," or ";" as a delimiter, but uses TAB. This is not recognized by Anymemo, and I had to put the comma by hand. There is no problem whatsoever with LibreOffice, as it asks for the delimiter to be used, and uses "," & ";" by default.

The csv-formatting which works best for me with Anymemo is

> 有, have
> 晚上, evening
> 菜, "cuisine, dish"

If a "," is used in the answer, it has to be _commented out_ for Anymemo not to be confused (obviously).
